tools/topology/v2/machine.conf

```diff
@@ -0,0 +1,80 @@
+# Instantiate a subtree - all the pipelines, connected to one DAI
+subtree {
```

Comment:
I would expect the machine file to contain
```
DeclarePipeline."pipe-name1" {
}

DeclarePipeline."pipe-name2" {
}

DeclareDAI."dai-name1" {
}
```
It should be representing the same data as contained in the machine M4 today.

Reply:
"Declare" in your code above means instantiation? I use "new." for it now. But yes, that's also what I do.
As far as I understand, the whole DSP audio topology can be divided into "DAI groups." I.e. each DAI can have a group of pipelines associated with it, but different DAIs never link. This seems to be what I see in most pipelines ATM but in fact I suspect this isn't a general rule? There can be some pass-through paths from a DMIC to a playback DAI, right? Do we have any such examples or any other examples, where different DAIs link together?
Currently in my notation a "subtree" is such an isolated part of the topology graph and it can include several pipelines, but only one DAI. The difference to the current m4 situation, where machine files just contain collections of pipelines, DAIs, backends, etc. is, that to make that approach work our m4 files make a heavy use of string-matching, i.e. concatenation. That's how DAIs, their configurations, graphs and pipelines are matched. To avoid that and to make explicit separation of graph parts I use "subtrees." E.g. each such subtree defines global parameters like `dai_format` etc. Without separating subtrees this would be impossible.